tensorflow/python/ops/metrics_impl.py

```diff
@@ -263,8 +263,9 @@ def mean(values, weights=None, metrics_collections=None,
       values = math_ops.multiply(values, weights)
       num_values = math_ops.reduce_sum(weights)
 
-    update_total_op = state_ops.assign_add(total, math_ops.reduce_sum(values))
-    update_count_op = state_ops.assign_add(count, num_values)
+    with ops.control_dependencies([values]):
+      update_total_op = state_ops.assign_add(total, math_ops.reduce_sum(values))
+      update_count_op = state_ops.assign_add(count, num_values)
```

Comment:
So this is for the else branch, above right? I think it's probably better placed there, because that's the one that introduces the race condition?